src/thunderbots/software/ai/navigator/rrt/rrt.cpp

```diff
@@ -0,0 +1,40 @@
+#include "rrt.h"
+#include "ai/intent/move_intent.h"
+#include "ai/primitive/move_primitive.h"
+
+RRTNav::RRTNav()
+{
+}
+
+std::vector<std::unique_ptr<Primitive>> RRTNav::getAssignedPrimitives(
+    const World &world, const std::vector<std::unique_ptr<Intent>> &assignedIntents) const
+{
+    std::vector<std::unique_ptr<Primitive>> assigned_primitives =
+        std::vector<std::unique_ptr<Primitive>>();
+
+    // Hand the different types of Intents here
+    for (const auto &intent : assignedIntents)
+    {
+        if (intent->getIntentName() == MOVE_INTENT_NAME)
+        {
+            // TODO: Implement this
+            // Cast down to the MoveIntent class so we can access its members
+            MoveIntent move_intent = dynamic_cast<MoveIntent &>(*intent);
+
+            std::unique_ptr<Primitive> move_prim =
```

Comment:
Would it be good to move logic specific to each primitive into its own function that takes a single Intent and returns a Primitive?